internal/cli/template_create.go

```diff
@@ -0,0 +1,184 @@
+package cli
+
+import (
+	"os"
+
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type ProjectTemplateCreateCommand struct {
+	*baseCommand
+
+	flagSummary                    string
+	flagExpandedSummary            string
+	flagReadmeMarkdownTemplatePath string
+	flagWaypointHCLTemplatePath    string
+	flagTFCNoCodeModuleSource      string
+	flagTFCNoCodeModuleVersion     string
+	flagTags                       []string
+}
+
+func (c *ProjectTemplateCreateCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+	ctx := c.Ctx
+
+	if len(args) != 1 {
+		c.ui.Output("Single argument required.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	name := args[0]
+
+	var template pb.ProjectTemplate
+
+	template.Name = name
+	if c.flagSummary != "" {
+		template.Summary = c.flagSummary
+	}
+	if c.flagExpandedSummary != "" {
+		template.ExpandedSummary = c.flagExpandedSummary
+	}
+
+	if c.flagReadmeMarkdownTemplatePath != "" {
+		rmt, err := os.ReadFile(c.flagReadmeMarkdownTemplatePath)
+		if err == os.ErrNotExist {
+			c.ui.Output("Readme template file does not exist: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		} else if err != nil {
+			c.ui.Output("Unable to read readme.md template file: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
```

Comment:
nit: because each of these errors ends in `return` we should decouple them (no need for `else` or `else if`). This generally improves readability and is more "idiomatic":

From [Control Structures](https://go.dev/doc/effective_go#control-structures)
> This is an example of a common situation where code must guard against a sequence of error conditions. The code reads well if the successful flow of control runs down the page, eliminating error cases as they arise. Since error cases tend to end in return statements, the resulting code needs no else statements.

Reply:
Optional refactor suggestion: 

```suggestion
		if err != nil {
		  // general error message
		  errMsg := "Unable to read readme.md template file: %s"
		  if err == os.ErrNotExist {
			  // give specific error message if the file does not exist
			  errMsg = "Readme template file does not exist: %s"
			} 
		
			c.ui.Output(errMsg, clierrors.Humanize(err), terminal.WithErrorStyle())
			return 1
		}
```